nucypher/characters/control/specifications/bob.py

```diff
@@ -27,11 +27,11 @@ class JoinPolicy(BaseSchema):  #TODO:  this doesn't have a cli implementation
     label = fields.Label(
         load_only=True, required=True,
         click=options.option_label(required=True))
-    alice_verifying_key = fields.Key(
+    publisher_verifying_key = fields.Key(
```

Comment:
Does the "Retreive" schema also need to be changed. Bob.retrieve(...) will call join policy as part of its operations. Currently the Retrieve schema only takes `alice_verifying_key`, but does it also need to take `publisher_verifying_key` since they can be two different keys?  See https://github.com/fjarri/nucypher/blob/chili-aftertaste/nucypher/characters/control/specifications/bob.py#L43 and https://github.com/fjarri/nucypher/blob/chili-aftertaste/nucypher/characters/control/interfaces.py#L222.

Reply:
For now we can leave only `alice_verifying_key`, `publisher` can be added later without breaking compatibility. But yes, eventually we will want both.

Reply:
Filed https://github.com/nucypher/nucypher/issues/2750.